File: PySnapScraper/check.py
#!/usr/bin/env python3
from datetime import datetime
import console
import ApiCalls
import Downloader 
import sys

class checking:
	def save_console(self, latitude, longitude, zoom):
		print("")
		print("Would you like to save the terminal output to a text file? y/n")
		saveLinks = console.userInput()

		print("Would you like to print the links? y/n")
		printLinks = console.userInput()

		if saveLinks and printLinks == "y" or saveLinks and printLinks == "Y" or saveLinks and printLinks == "yes":
			dateTime = datetime.now()
			filename = "SnapScraperConsole-" + str(dateTime) + '.txt'

			with open(filename, 'w') as f:
				sys.stdout = f

				def intializer(latitude, longitude, zoom):
					epoch = ApiCalls.SCApiCalls.getEpoch(latitude, longitude, zoom)
					(mediaURLS, mediaDictionary) = ApiCalls.SCApiCalls.getPlaylist(latitude, longitude, zoom, epoch, True)

					checker(longitude, latitude, mediaURLS, mediaDictionary)

					Downloader.Downloader.downloadFiles(latitude, longitude, mediaURLS, mediaDictionary)
				
				def checker(longitude, latitude, urlArray, urlDictionary): 	
					print("")
					print("Would you like to print the links? y/n")
					printLinks = 'y'

					if len(urlArray) == 0:
						print("No Media URLs exist.")
					elif printLinks == "y" or printLinks == "Y" or printLinks == "yes":
						print("")
						for item in urlArray:
							print(item)

				intializer(latitude, longitude, zoom)

				console.finish()
				#sys.stdout.close()

		elif (saveLinks == "y" or saveLinks == "Y" or saveLinks == "yes") and (printLinks == 'n' or printLinks == "N" or printLinks == "no"):
			dateTime = datetime.now()
			filename = "SnapScraperConsole-" + str(dateTime) + '.txt'

			with open(filename, 'w') as f:
				sys.stdout = f

				def intializer(latitude, longitude, zoom):
					epoch = ApiCalls.SCApiCalls.getEpoch(latitude, longitude, zoom)
					(mediaURLS, mediaDictionary) = ApiCalls.SCApiCalls.getPlaylist(latitude, longitude, zoom, epoch, True)
					
					Downloader.Downloader.downloadFiles(latitude, longitude, mediaURLS, mediaDictionary)

				intializer(latitude, longitude, zoom)

				console.finish()
				#sys.stdout.close()

		else:
			def intializer(latitude, longitude, zoom):
					epoch = ApiCalls.SCApiCalls.getEpoch(latitude, longitude, zoom)
					(mediaURLS, mediaDictionary) = ApiCalls.SCApiCalls.getPlaylist(latitude, longitude, zoom, epoch, True)

					checker(longitude, latitude, mediaURLS, mediaDictionary)

					Downloader.Downloader.downloadFiles(latitude, longitude, mediaURLS, mediaDictionary)

				
			def checker(longitude, latitude, urlArray, urlDictionary): 
				print("")
				printLinks = 'y'

				if len(urlArray) == 0:
					print("No Media URLs exist.")
				elif printLinks == "y" or printLinks == "Y" or printLinks == "yes":
					print("")
					for item in urlArray:
						print(item)

			intializer(latitude, longitude, zoom)
			console.finish()


	#sys.stdout.close()		


# choice = checking()
# choice.save_console(-34.92220775250722, 138.60374304787342, 14.743706116624152)

#intializer(-34.92220775250722, 138.60374304787342, 14.743706116624152)

#https://stackabuse.com/writing-to-a-file-with-pythons-print-function/

File: Snap-Scraper/Check.swift
//
//  check.swift
//  SCMapDownloader
//
//  Created by Richard Matthews on 15/5/20.
//  Copyright © 2020 Richard Matthews. All rights reserved.
//
import Foundation

class Check {
    
    let caller = SCApiCalls()
    let downloader = Downloader()
    
    func checker(lat:String, long:String, zoom:String, consoleIO:ConsoleIO){
        let epoch = caller.getEpoch(lat: lat,lon: long,zoom: zoom)
        print("EPOCH IS:", epoch)
        let (mediaURLS, mediaDictionary) = caller.getPlaylist(lat: lat,lon: long,zoom: zoom,epoch: epoch,flag:"checker")
        
        //exportToLog()
        
        print("")
        print("Would you like to print the links? y/n")
        let printLinks = consoleIO.getInput()
        
        switch printLinks {
            
        case "y":
            print("")
            // loop through the mediaURLS and print them to the screen on a newline
            for item in mediaURLS {
                print("\(item)")
            }
        default:
            break
        }
        print("")
        print("Would you like to save the links to file? y/n")
        let saveLinks = consoleIO.getInput()
        
        switch saveLinks {
            
        case "y":
            print("")
            // loop through the mediaURLS and print them to the screen on a newline
            // Set the file path
            let date = Date()
            let format = DateFormatter()
            format.dateFormat = "yyyy-MM-dd HH:mm:ss"
            let formattedDate = format.string(from: date)
            let documentsURL = FileManager.default.urls(for: .downloadsDirectory, in: .userDomainMask)[0]
            
            let path = documentsURL.appendingPathComponent("Snapchat/\(formattedDate)-@lat\(lat)-lon\(long)-listing.txt")
            
            let output = mediaURLS.joined(separator: "\n")
            
            do {
                try output.write(to: path, atomically: false, encoding: String.Encoding.utf8)
                //print("\(output)")
            }
            catch let error as NSError {
                print(error.localizedDescription)
                print("Ooops! Something went wrong: \(error)")
            }
        default:
            break
        }
            print("")
            print("Would you like to download the media? y/n")
            let downloadLinks = consoleIO.getInput()
            
            switch downloadLinks {
                
            case "y":
                print("")
                downloader.downloadFile(lat:lat, long:long, urlArray:mediaURLS, urlDictionary: mediaDictionary)
            default:
                print("")
                break
            }
    
    }

    
}

